Objective-C/CBLReplicator.mm

```diff
@@ -291,14 +294,24 @@ - (void) _start {
             convertError(err, &error);
             CBLWarnError(Sync, @"%@: Replicator cannot be created: %@",
                          self, error.localizedDescription);
-            return;
+            *c4err = err;
+            return nil;
         }
-        c4repl_start(_repl);
+    }
+    
+    return _repl;
+}
+
+- (void) _start {
+    C4Error err;
+    C4Replicator* repl = [self c4repl: &err];
```

Comment:
Please do warning the error and return here if repl is nil. You can move CBLWarnError from c4repl: method here.